EasyLayout.Forms/EasyLayout.cs

```diff
@@ -567,112 +548,52 @@ private static int ConstantValue(Expression expr)
 
         private static bool IsConstant(Expression expr)
         {
-            if (expr.NodeType == ExpressionType.Constant)
+            switch (expr)
             {
-                return true;
-            }
-
-            if (expr.NodeType == ExpressionType.MemberAccess)
-            {
-                var mexpr = (MemberExpression)expr;
-                // todo: Does this even work, where the heck is MemberType?
-                if (mexpr.NodeType == ExpressionType.Constant)
-                {
+                case ConstantExpression _:
                     return true;
-                }
-                return false;
-            }
-
-            if (expr.NodeType == ExpressionType.Convert)
-            {
-                var cexpr = (UnaryExpression)expr;
-                return IsConstant(cexpr.Operand);
-            }
-
-            var methodCall = expr as MethodCallExpression;
-            if (methodCall != null)
-            {
-                return methodCall.Method.Name == nameof(ToConst);
+                case MemberExpression mexpr:
+                    // todo: Does this even work, where the heck is MemberType?
+                    return mexpr.NodeType == ExpressionType.Constant;
+                case UnaryExpression cexpr:
+                    return IsConstant(cexpr.Operand);
+                case MethodCallExpression methodCall:
+                    return methodCall.Method.Name == nameof(ToConst);
+                default:
+                    return false;
             }
-
-            return false;
         }
 
         private static LeftExpression ParseLeftExpression(Expression expr)
         {
-            Position? position = null;
-            MemberExpression viewExpr = null;
-
             var fExpr = expr as MethodCallExpression;
-            if (fExpr != null)
-            {
-                position = GetPosition(fExpr);
-                viewExpr = fExpr.Arguments.FirstOrDefault() as MemberExpression;
-            }
-
-            if (position == null)
-            {
-                var memExpr = expr as MemberExpression;
-                if (memExpr == null)
-                {
-                    throw new NotSupportedException("Left hand side of a relation must be a member expression, instead it is " + expr);
```

Comment:
I added a unit test project and a handful of negative unit tests to master to help ensure we don't lose functionality.  I'll try to add some more over the weekend.